addons/hr_timesheet/report/timesheets_analysis_report.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from psycopg2 import sql
+
+from odoo import api, tools, fields, models
+
+
+class TimesheetsAnalysisReport(models.Model):
+    _name = "timesheets.analysis.report"
+    _description = "Timesheets Analysis Report"
+    _auto = False
+
+    name = fields.Char("Description", readonly=True)
+    user_id = fields.Many2one("res.users", string="User", readonly=True)
+    project_id = fields.Many2one("project.project", string="Project", readonly=True)
+    task_id = fields.Many2one("project.task", string="Task", readonly=True)
+    ancestor_task_id = fields.Many2one("project.task", string="Ancestor Task", readonly=True)
+    employee_id = fields.Many2one("hr.employee", string="Employee", readonly=True)
+    manager_id = fields.Many2one("hr.employee", "Manager", readonly=True)
+    company_id = fields.Many2one("res.company", string="Company", readonly=True)
+    department_id = fields.Many2one("hr.department", string="Department", readonly=True)
+    currency_id = fields.Many2one(related="company_id.currency_id", string="Currency", readonly=True)
```

Comment:
We should avoid having related as there is a depends associated with it. It is better to add a join and get it in the report directly.